core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -1300,6 +1300,67 @@ class ReplicaManagerTest {
     TestUtils.assertNoNonDaemonThreads(this.getClass.getName)
   }
 
+  @Test
+  def testHasPreferredReplica(): Unit = {
+    val topicPartition = 0
+    val topicId = Uuid.randomUuid()
+    val followerBrokerId = 0
+    val leaderBrokerId = 1
+    val leaderEpoch = 1
+    val leaderEpochIncrement = 2
+    val countDownLatch = new CountDownLatch(1)
+
+    // Prepare the mocked components for the test
+    val props = new Properties()
+    props.put(KafkaConfig.ReplicaSelectorClassProp, "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+    val (replicaManager, _) = prepareReplicaManagerAndLogManager(new MockTimer(time),
+      topicPartition, leaderEpoch + leaderEpochIncrement, followerBrokerId,
+      leaderBrokerId, countDownLatch, expectTruncation = true, topicId = Some(topicId), extraProps = props)
```

Comment:
Using this method create a lot of unnecessary noise in my opinion. Nowadays, we tend to use `setupReplicaManagerWithMockedPurgatories` which is simpler. `setupReplicaManagerWithMockedPurgatories` does not support racks though. I suppose that we could it accept an optional Map from broker id to rack. Could we try using this one instead? There are many example in this file.

Reply:
@dajac  Thanks for your review，I made changes for the test and add some comment.